File: src/UI/PiBooking.UI/src/app/models/OrderViewModel.ts

import { CustomerViewModel } from './CustomerViewModel'
import { JobViewModel } from './JobViewModel'
import { TimeSlotViewModel } from './TimeSlotViewModel'
import { PaymentViewModel } from './PaymentViewModel'
import { EngineerViewModel } from './EngineerViewModel'

export class OrderViewModel {
    public customer: CustomerViewModel;
    public job: JobViewModel;
    public timeSlots: TimeSlotViewModel[];
    public payment: PaymentViewModel;
    public totalAmount: number; //one way to server
    public totalCount: number;
    public signature: string;
    public engineer: EngineerViewModel = null;
    public orderID: number;
    public total: number //one way from server

    constructor() {

        this.customer = new CustomerViewModel();
        this.job = new JobViewModel();
        this.timeSlots = new Array<TimeSlotViewModel>();
        this.payment = new PaymentViewModel();
        this.totalCount = 0;
        this.totalAmount = 0;
    }
}

File: src/PiBooking.API/ViewModels/OrderViewModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PiBooking.Core.ViewModels
{
    public class OrderViewModel
    {
        public int OrderID { get; set; }
        public CustomerViewModel Customer { get; set; }
        public JobViewModel Job { get; set; } 
        public EngineerViewModel Engineer { get; set; }
        public bool HasPaid { get; set; }
        public List<TimeSlotViewModel> TimeSlots { get; set; }
        public string Signature { get; set; }
         
        public Decimal  Total { get; set; }

        public DateTime? CreateDate { get; set; }

        public DateTime? UpdateDate { get; set; }
    }
}
